src/SDIOAnalyzer.cpp

```diff
@@ -176,59 +176,70 @@ void SDIOAnalyzer::PacketStateMachine()
 //  - Long Response
 //  - Data
 
-U32 SDIOAnalyzer::FrameStateMachine( void )
+bool SDIOAnalyzer::FrameStateMachine( void )
 {
+    Frame frame;
+    bool done = false;
+    U8 respLength;
+
     switch( frameState )
     {
     case TRANSMISSION_BIT:
-    {
-        Frame frame;
         frame.mStartingSampleInclusive = lastFallingClockEdge;
-        frame.mEndingSampleInclusive = mClock->GetSampleOfNextEdge() - 1;
+        frame.mEndingSampleInclusive = mClock->GetSampleOfNextEdge();
         frame.mFlags = 0;
         frame.mData1 = mCmd->GetBitState();
         frame.mType = FRAME_DIR;
         mResults->AddFrame( frame );
 
-        FrameV2 frame_v2;
-        frame_v2.AddBoolean( "DIR", frame.mData1 );
-        mResults->AddFrameV2( frame_v2, "DIR", frame.mStartingSampleInclusive, frame.mEndingSampleInclusive );
+        frameV2 = new FrameV2;
```

Comment:
Since the conversation got clobbered due the the update:
>It needs to be pointer because, I want to reset the data to the state after construction. I tried with a unique pointer, but I cannot delete it manually. The problem is that FrameStateMachine is re-entrant, so I do not want to lose the data after going out of scope. On the other hand I want to be able to reset the data, but Saleae did not implement the rule of three for FrameV2. A raw pointer seemed to me the best fit, with all its caveats.

I believe the only changes that are necessary is to `#include <memory>`, update the type in the header, make this line `frameV2 = std::unique_ptr(new FrameV2)`, and call `frameV2.reset()` instead of `delete frameV2`. Is there something I'm not seeing?

My mind is a little slow this morning. Could you spell out the implications of missing a copy assignment operator and a copy constructor for FrameV2?

Reply:
Ah, reset() :facepalm:. I will try this.

Reply:
With reset, Logic crashes.